### DISCUSSION TITLE
An unhandled win32 exception occurred in...

### DISCUSSION THREAD

**Discussion by mehmetilker** [dotnet/maui#8462] [Q&A]
Sometimes I am seeing following exception and neither can see details or catch the exception in code.

Sometimes it happend when I close the app. For example what I am doing a rediction to xaml page from razor page with following way:

```c#
    async void GoToCategory()
    {
        try
        {
            await Shell.Current.GoToAsync("//category/world?CategoryId=world",
                      new Dictionary<string, object>
                          {
                              ["bear"] = "aaa"
                          });
        }
        catch(Exception ex)
        {
            Debug.WriteLine(ex.ToString());
        }      
    }
```

Is there a way to see details of this kind of exceptions?

![image](https://user-images.githubusercontent.com/279289/176872754-acee6b0d-fba9-47b4-b9d5-822545692336.png)

**Comment by mehmetilker** [comment]
I could find some details in Event Viewer. From Exception code: 0xc0000005 it looks similar to https://github.com/dotnet/maui/issues/8372
Works on Android emulator but not on Windows.

```
Faulting application name: MauiApp1.exe, version: 1.0.0.0, time stamp: 0x62571213
Faulting module name: Microsoft.UI.Input.dll, version: 10.0.22621.1000, time stamp: 0x5878828b
Exception code: 0xc0000005
Fault offset: 0x000000000003f1f5
Faulting process ID: 0x56f4
Faulting application start time: 0x01d88e136a5dedb7
Faulting application path: C:\Z\X.App\MauiApp1\bin\Debug\net6.0-windows10.0.19041.0\win10-x64\AppX\MauiApp1.exe
Faulting module path: C:\Program Files\WindowsApps\Microsoft.WindowsAppRuntime.1.1_1000.516.2156.0_x64__8wekyb3d8bbwe\Microsoft.UI.Input.dll
Report ID: 862259fe-fdc6-4e07-9747-8e863a22bed8
Faulting package full name: 44B11EC2-D8C2-4E17-BAAC-8A5E4483BF47_1.0.0.1_x64__9zz4h110yvjzm
Faulting package-relative application ID: App
```

**Reply by mnasers77** [reply]
I am also seeing this issue on Windows. It appears to happen when you try to call Shell.Current.GoToAsync( ) while another page is being navigated to but is still not visible. I see the issue when I try to auto login a user who has previously logged in. I found this sample project on YouTube when trying to debug my issue. It has similar functionality and creates the same error when I run it on Windows. https://github.com/mistrypragnesh40/SimpleLoginUI.git

**Comment by 8m0** [comment]
I have the same issue on Windows. Error occurs on `await Shell.Current.GoToAsync($"//{nameof(APage)}");`. It's just basic navigation after checking if user is logged in for an App onload.

**Comment by Mzazvor** [comment]
Im having the same issue as @mehmetilker. My app works on android emulator and local devices. Crash happens exactly like @8m0 describes.

**Comment by amo-sport-tech** [comment]
Yep, struggling with the same issue. If it place a button on the xaml and use it after the page has loaded then it works, if i try to navigate away during navigation itself, the crash occurs. We have a loading page on app load for example to dynamically decide where to take the user depending on if they are logged in or not. Have tried using OnAppearing/OnNavigatedTo and neither work.

**Comment by GasimoCodes** [comment]
Same issue on my MAUI (6), on the official demo I attempt to do a check and navigate in the default ContentPage constructor and get thrown a Win32 exception, doesnt happen when I put the same code under a button as @amo-sport-tech noted. 
(The order of InitComponent and Shell.Current.GoTo does not have any impact on the exception occuring)

```
public MainPage()
{
      InitializeComponent();
      Shell.Current.GoToAsync("//WelcomePage", true);
}
```

**Comment by GasimoCodes** [comment]
For anyone stumbling onto this discussion, I have opened an issue over at #12038
